Detectors/TPC/reconstruction/include/TPCReconstruction/BoxClusterer.h

```diff
@@ -19,82 +19,98 @@
-      CalPad*   mPedestals;    //!<! Pedestal data
-
-      std::vector<o2::TPC::Cluster>* mClusterArray; ///< Internal cluster storage
-
-      ClassDefNV(BoxClusterer, 1);
-    };
-  }
+namespace TPC
+{
+
+class ClusterContainer;
+
+class BoxClusterer : public Clusterer
+{
+
+  using MCLabelContainer = o2::dataformats::MCTruthContainer<o2::MCCompLabel>;
+
+ public:
+  /// Constructor
+  /// \param output is pointer to vector to be filled with clusters
+  /// \param rowsMax Max number of rows to process
+  /// \param padsMax Max number of pads to process
+  /// \param timeBinsMax Max number of timebins to process
+  /// \param minQMax Minimum peak charge for cluster
+  /// \param requirePositiveCharge Positive charge is required
+  /// \param requireNeighbouringPad Requires at least 2 adjecent pads with charge above threshold
+  BoxClusterer(std::vector<o2::TPC::Cluster>* output,
+               int rowsMax = 18,
+               int padsMax = 138,
+               int timeBinsMax = 1024,
+               int minQMax = 5,
+               bool requirePositiveCharge = true,
+               bool requireNeighbouringPad = true);
+
+  /// Destructor
+  ~BoxClusterer();
+
+  /// Steer conversion of points to digits
+  /// \param digits Container with TPC digits
+  /// \param mcDigitTruth MC Digit Truth container
+  /// \param eventCount event counter
+  void Process(std::vector<o2::TPC::Digit> const& digits, MCLabelContainer const& mcDigitTruth, int eventCount) override;
+  void FinishProcess(std::vector<o2::TPC::Digit> const& digits, MCLabelContainer const& mcDigitTruth, int eventCount) override;
+
+  void setRowsMax(int val) { mRowsMax = val; };
+  void setPadsMax(int val) { mPadsMax = val; };
+  void setTimeBinsMax(int val) { mTimeBinsMax = val; };
+  void setMinQMax(float val) { mMinQMax = val; };
+  void setRequirePositiveCharge(bool val) { mRequirePositiveCharge = val; };
+  void setRequireNeighbouringPad(bool val) { mRequireNeighbouringPad = val; };
+
+  int getRowsMax() const { return mRowsMax; };
+  int getPadsMax() const { return mPadsMax; };
+  int getTimeBinsMax() const { return mTimeBinsMax; };
+  float getMinQMax() const { return mMinQMax; };
+  bool hasRequirePositiveCharge() const { return mRequirePositiveCharge; };
+  bool hasRequireNeighbouringPad() const { return mRequireNeighbouringPad; };
+
+ private:
+  // To be done
+  /* BoxClusterer(const BoxClusterer &); */
+  /* BoxClusterer &operator=(const BoxClusterer &); */
+
+  void FindLocalMaxima(const Int_t iCRU);
+  void CleanArrays();
+  void GetPadAndTimeBin(Int_t bin, Short_t& iPad, Short_t& iTimeBin);
+  Int_t Update(const Int_t iCRU, const Int_t iRow, const Int_t iPad,
+               const Int_t iTimeBin, Float_t signal);
+  Float_t GetQ(const Float_t* adcArray, const Short_t pad,
+               const Short_t time, Short_t& timeMin, Short_t& timeMax,
+               Short_t& padMin, Short_t& padMax) const;
+  Bool_t UpdateCluster(Float_t charge, Int_t deltaPad, Int_t deltaTime,
```

Comment:
Since you are changing the classes it would be nice to change the function names to lower case in order to comply the naming conventions. There are more places in the other classes. Please have a look.

Reply:
yes, they are all somehow related. 
and yes, then I can also change the function names.

Currently I'm still trying to fix some broken macros (and with that introducing again the second type of cluster output container, vector<Cluster> in addition to vector<ClusterHardwareContainer8kb>).

When all this is done, I will remove the [WIP].